sdks/python/generate_pydoc.sh

```diff
@@ -31,43 +31,88 @@ rm -rf target/docs/*
 
 mkdir -p target/docs/source
 
-# Exclude internal/experimental files from the documentation.
+# Sphinx apidoc autodoc options
+export SPHINX_APIDOC_OPTIONS=\
+members,\
+undoc-members,\
+show-inheritance
+
+# Exclude internal/experimental/Cython paths/patterns from the documentation.
 excluded_internal_code=(
+    apache_beam/coders/stream.py*
     apache_beam/examples/
-    apache_beam/internal/clients/
-    apache_beam/io/gcp/internal/clients/
+    apache_beam/internal/
+    apache_beam/io/gcp/internal/
+    apache_beam/io/gcp/tests/
     apache_beam/runners/api/
     apache_beam/runners/test/
+    apache_beam/runners/dataflow/internal/
     apache_beam/runners/portability/
     apache_beam/runners/worker/
-    apache_beam/runners/dataflow/internal/clients/
-    apache_beam/testing/data/)
-
-python $(type -p sphinx-apidoc) -f -o target/docs/source apache_beam \
-    "${excluded_internal_code[@]}" "*_test.py"
+    apache_beam/testing/
+    *_pb2.py
+    *_test.py
+    *_test_common.py
+    *_test_utils.py
+)
 
-# Remove Cython modules from doc template; they won't load
-sed -i -e '/.. automodule:: apache_beam.coders.stream/d' \
-    target/docs/source/apache_beam.coders.rst
+python $(type -p sphinx-apidoc) -fMeT -o target/docs/source apache_beam \
+    "${excluded_internal_code[@]}"
 
-# Create the configuration and index files
+#=== conf.py ===#
 cat > target/docs/source/conf.py <<'EOF'
 import os
 import sys
 
+import sphinx_rtd_theme
+
 sys.path.insert(0, os.path.abspath('../../..'))
 
+exclude_patterns = [
+    '_build',
+    'target/docs/source/apache_beam.rst',
+]
+
 extensions = [
     'sphinx.ext.autodoc',
+    'sphinx.ext.doctest',
+    'sphinx.ext.intersphinx',
     'sphinx.ext.napoleon',
     'sphinx.ext.viewcode',
 ]
 master_doc = 'index'
-html_theme = 'sphinxdoc'
+html_theme = 'sphinx_rtd_theme'
+html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 project = 'Apache Beam'
+
+autoclass_content = 'both'
+autodoc_member_order = 'bysource'
+
+doctest_global_setup = '''
+import apache_beam as beam
+import sys
+'''
+
+intersphinx_mapping = {
+  'python': ('https://docs.python.org/2', None),
+}
+
+nitpicky = True
+nitpick_ignore = [
+  # Ignore internal private classes. If there's any reference to them within the
+  # documentation, it will report as a broken cross reference. This can happen
+  # if a class inherits from a private class.
+  ('py:class', 'apache_beam.coders.coders._PickleCoderBase'),
+  ('py:class', 'apache_beam.io.gcp.datastore.v1.datastoreio._Mutate'),
+  ('py:class', 'apache_beam.runners.direct.executor.CallableTask'),
+  ('py:class', 'apache_beam.transforms.trigger._ParallelTriggerFn'),
+]
 EOF
+
+#=== index.rst ===#
 cat > target/docs/source/index.rst <<'EOF'
-.. include:: ./modules.rst
+.. include:: ./apache_beam.rst
```

Comment:
Ah, all the .rst files are generated by ``sphinx-apidoc``. That is then used as input for ``sphinx`` to generate the .html files.